A large number of small changes to improve readability and maintainability

kwhitefoot opened the pull request:
I'm using this project as a way of becoming more familiar with both Android and Java.  

My personal way of doing this is to do a code review exercise.

I have made a lot of small changes all aimed at making the code easier to read and safer to maintain.  The largest change is the introduction of an enum called Security that allows the switch statements that define proto and acceptcrt to be eliminated along with several other if statements.  I have tried to use strong typed fields and methods instead of using strings.

It seems to work but I can't claim to have tested it extensively.

Anyway, if you think it is useful here it is.

kwhitefoot committed:
Added comments to make it easier to understand

The process of adding comments to code that I read helps me make sense of it even if I never read them again.  Comments that are just a question mark are place holders to return to when I have a better idea of what that piece of code is for.

kwhitefoot committed:
Added comments.

kwhitefoot committed:
Reformatted all files to get consistent indentation, etc.

kwhitefoot committed:
Corrected Accont -> Account.

Added Security Enum, WIP to get rid of literal code numbers in configuration and replace with enum names.

kwhitefoot committed:
Added proto and acceptcrt properties to Security enum so that there is no longer a need for a switch statement to find them.

kwhitefoot committed:
Folder override can never be null so do not check for it.

kwhitefoot committed:
Add notes to README.md to explain the reason for the fork and briefly describe what has been done.

nbenm commented:
Thanks, very good idea.
I’m a little bit busy, and I’ll be on leave this week.
I’ll try to have a look at your changes next week

Noury


> Le 5 nov. 2016 à 01:40, kwhitefoot <notifications@github.com> a écrit :
> 
> I'm using this project as a way of becoming more familiar with both Android and Java.
> 
> My personal way of doing this is to do a code review exercise.
> 
> I have made a lot of small changes all aimed at making the code easier to read and safer to maintain. The largest change is the introduction of an enum called Security that allows the switch statements that define proto and acceptcrt to be eliminated along with several other if statements. I have tried to use strong typed fields and methods instead of using strings.
> 
> It seems to work but I can't claim to have tested it extensively.
> 
> Anyway, if you think it is useful here it is.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/nbenm/ImapNote2/pull/28
> 
> Commit Summary
> 
> 	• Added comments to make it easier to understand
> 	• Added comments.
> 	• Reformatted all files to get consistent indentation, etc.
> 	• Corrected Accont -> Account.
> 	• Added proto and acceptcrt properties to Security enum so that there is no longer a need for a switch statement to find them.
> 	• Folder override can never be null so do not check for it.
> File Changes
> 
> 	• M ImapNote2/build.gradle (4)
> 	• M ImapNote2/src/main/AndroidManifest.xml (152)
> 	• D ImapNote2/src/main/java/com/Pau/ImapNotes2/AccontConfigurationActivity.java (349)
> 	• A ImapNote2/src/main/java/com/Pau/ImapNotes2/AccountConfigurationActivity.java (356)
> 	• M ImapNote2/src/main/java/com/Pau/ImapNotes2/Data/ConfigurationFile.java (255)
> 	• M ImapNote2/src/main/java/com/Pau/ImapNotes2/Data/ImapNotes2Account.java (88)
> 	• M ImapNote2/src/main/java/com/Pau/ImapNotes2/Data/NotesDb.java (157)
> 	• M ImapNote2/src/main/java/com/Pau/ImapNotes2/ImapNotes2.java (102)
> 	• M ImapNote2/src/main/java/com/Pau/ImapNotes2/Listactivity.java (458)
> 	• M ImapNote2/src/main/java/com/Pau/ImapNotes2/Miscs/ImapNotes2Result.java (26)
> 	• M ImapNote2/src/main/java/com/Pau/ImapNotes2/Miscs/ImapNotesAuthenticatorService.java (159)
> 	• M ImapNote2/src/main/java/com/Pau/ImapNotes2/Miscs/Imaper.java (286)
> 	• M ImapNote2/src/main/java/com/Pau/ImapNotes2/Miscs/OneNote.java (106)
> 	• M ImapNote2/src/main/java/com/Pau/ImapNotes2/Miscs/Sticky.java (78)
> 	• M ImapNote2/src/main/java/com/Pau/ImapNotes2/Miscs/SyncThread.java (34)
> 	• M ImapNote2/src/main/java/com/Pau/ImapNotes2/Miscs/UpdateThread.java (137)
> 	• M ImapNote2/src/main/java/com/Pau/ImapNotes2/NewNoteActivity.java (72)
> 	• M ImapNote2/src/main/java/com/Pau/ImapNotes2/NoteDetailActivity.java (266)
> 	• M ImapNote2/src/main/java/com/Pau/ImapNotes2/NotesListAdapter.java (84)
> 	• A ImapNote2/src/main/java/com/Pau/ImapNotes2/Sync/Security.java (75)
> 	• M ImapNote2/src/main/java/com/Pau/ImapNotes2/Sync/SyncAdapter.java (120)
> 	• M ImapNote2/src/main/java/com/Pau/ImapNotes2/Sync/SyncUtils.java (857)
> 	• M ImapNote2/src/main/res/layout/account_selection.xml (354)
> 	• M ImapNote2/src/main/res/layout/main.xml (12)
> 	• M ImapNote2/src/main/res/layout/new_note.xml (12)
> 	• M ImapNote2/src/main/res/layout/note_detail.xml (41)
> 	• M ImapNote2/src/main/res/layout/note_element.xml (45)
> 	• M ImapNote2/src/main/res/menu/detail.xml (54)
> 	• M ImapNote2/src/main/res/menu/list.xml (37)
> 	• M ImapNote2/src/main/res/menu/newnote.xml (8)
> 	• M ImapNote2/src/main/res/values-v14/styles.xml (29)
> 	• M ImapNote2/src/main/res/values/strings.xml (5)
> 	• M ImapNote2/src/main/res/values/styles.xml (25)
> 	• M ImapNote2/src/main/res/xml/authenticator.xml (9)
> 	• M ImapNote2/src/main/res/xml/searchable.xml (4)
> 	• M ImapNote2/src/main/res/xml/syncadapter.xml (7)
> 	• M build.gradle (2)
> 	• M gradle/wrapper/gradle-wrapper.properties (4)
> Patch Links:
> 
> 	• https://github.com/nbenm/ImapNote2/pull/28.patch
> 	• https://github.com/nbenm/ImapNote2/pull/28.diff
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

kwhitefoot committed:
Start moving arguments from execute to constructor to avoid the need to cast.  Moved notes.org to root.

kwhitefoot committed:
Created ConfigurationFieldNames class to name the fields in the config. file.

kwhitefoot committed:
Fixed bug introduced by changes.  Bug was incorrect casting of useSticky.

kwhitefoot committed:
Make patterns for analysing sticky notes static final so that we do not reuse variables and so that we do not compile them every time.

kwhitefoot committed:
Made notesFolder local instead of argument in DeleteNote.

kwhitefoot committed:
Move arguments from execute to constructor of UpdateThread to enforce type safety.

kwhitefoot committed:
Ran Field can be local analysis and fixed almost all.

kwhitefoot committed:
ConstantConditions analyser.  Some possible null references remain.  Moved code inside try in some cases, use Log.d instead of exception.printStackTrace.  Suppress MessagingException when closing connection because the connection is always closed after the call even if the execption is thrown.

kwhitefoot committed:
Code Maturity analysis.

kwhitefoot committed:
Pass arguments via constructor instead of execute; thi means that we do not need to cast or convert.

kwhitefoot committed:
Removed all unused imports.

kwhitefoot committed:
Removed redundant local variables.

kwhitefoot committed:
Use class instead of instance to refer to static fields.

kwhitefoot committed:
Accepted all Declaration access can be weaker suggestions.

kwhitefoot committed:
Applied fix Make Final to all suggested lines.

kwhitefoot committed:
Removed all redundant throws.

kwhitefoot committed:
Use <> instead of explicit type arguments.

kwhitefoot committed:
Fixed empty XML elements.

kwhitefoot committed:
Oops, forgot the progree dialog in doInBackground.  Fixed.

kwhitefoot committed:
Added intent item names to make it easier to understand and enforce proper comunication between activities.

kwhitefoot committed:
Put all the sticky colour related values together in the Colors enum.

kwhitefoot committed:
Removed unused constructor from Sticky.

kwhitefoot committed:
Added comments to manifest to ask why we have met-data elements.

kwhitefoot committed:
Use strings resource instead of literal strings to enable translation.

kwhitefoot committed:
Ran the Infer nullity analyser to annotate the code with @Nullable to provide hints for possible improvements to the code to reduce the risk of null reference errors.

kwhitefoot committed:
Removed Imaper.sfolder because it was only assigned but never read.

kwhitefoot committed:
Fixed bug associated with background colour.  None should have White not zero.

kwhitefoot committed:
Added Utilities class.  Only members so far are to define the internal date format.

kwhitefoot committed:
Use try with resource to ensure that cursors are closed after use.

kwhitefoot committed:
No more warnings in NotesDb!

kwhitefoot committed:
Only unused method warnings left in NotesListAdapter.

kwhitefoot committed:
Almost all warning fixed in UpdateThread.

kwhitefoot committed:
Made StubProvider package local.  Added TODO reminder to find out what it is for.  Marked methods as @Nullable because they return null.

kwhitefoot committed:
AndroidDomInspection suppressed,

kwhitefoot committed:
Most literal strings have now been replaced with string resources.

kwhitefoot committed:
Partial translation to Norwegian (bokmal)

kwhitefoot committed:
Moved classes to different packages to remove cyclic dependencies.

kwhitefoot committed:
Moved classes to different packages to remove cyclic dependencies.

kwhitefoot committed:
All warnings fixed or suppressed with explanations.

kwhitefoot committed:
Initialize security field to None so that we can ensure NonNull.
Added enumAction to UpdateThread so that we do not need to use strings.

kwhitefoot committed:
Created function for creating a progress dialog, shortens the code.  Converted more strings to resources.

kwhitefoot committed:
Deleted unused arguments.

kwhitefoot committed:
Renamed SyncAdapter.context to applicationContext to make it clear that the singleton app context is required here.  Renamed accountname to accountName.

kwhitefoot committed:
Use constants for column names instead of literal strings.

kwhitefoot committed:
Use File constructor instead of string concatenation to create file and directory paths.

kwhitefoot committed:
Added log messages.

kwhitefoot committed:
Created a function to create an intent to notify the listview of sync complete instead of repeating the code three times.

kwhitefoot committed:
Ran Android Lint and fixed a few minor complaints.

kwhitefoot committed:
Acted on all Declaration Redundancy issues.

kwhitefoot committed:
Acted on and suppressed some probable bugs reported by the analyser.

kwhitefoot committed:
Make res local because it was only the error message that needed to be passed to the post event.

kwhitefoot committed:
Ensure that security can never be null.

kwhitefoot committed:
Various possible error reports fixed.

kwhitefoot committed:
Put repeated code into a function.

kwhitefoot committed:
Add functions to remove the need for local variables.

kwhitefoot committed:
Removed unused position argument.  Added javadoc comments.  Corrected spelling of ReadStickyNote.  Reinstated onClick even handler that the analyser said was unused.

kwhitefoot committed:
Announce that future changes will probably break compatibility with ImapNotes2.

kwhitefoot committed:
Split GetOneNote into SaveNote and SaveNoteAndUpdatDatabase so that we do not need the updateDB argument.  This argument was a literal in all calls which means that the caller already knew statically exactly which part of the function was needed so the decision need not be checked again at run time.

Mystro256 commented:
I would start swashing related commits if I were you. Its a bit hard to review in this state due to the numerous commits. For example, all the commits that add comments to the code can be swashed together.

kwhitefoot commented:
It never occurred to me that anyone would really be interested enough to make it worthwhile to be more organised.  The fork is really just for me to use as a learning exercise. So far all of the changes are simply tidying up the code to reduce the risk of things like null references, uncaught exceptions, etc. , and to introduce enumerated types and named constants to reduce the risk of typos.

Now that someone has shown an interest I'll make an effort to make the changes in a more readable fashion.

One question, "swashing": is this some git related term?  I'm far from being an expert git user, I normally use tfs in my professional life.

On 29 Nov 2016 14:52, "Jeremy Newton" <notifications@github.com<mailto:notifications@github.com>> wrote:

I would start swashing related commits if I were you. Its a bit hard to review in this state due to the numerous commits. For example, all the commits that add comments to the code can be swashed together.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/nbenm/ImapNote2/pull/28#issuecomment-263575048>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AALNd2wcRanf-8sHfLV9YeVu0DZtWQIWks5rDC4LgaJpZM4KqHOu>.

Mystro256 commented:
I don't see why not, improving opensource code is always a good thing!
Also sorry, I meant to say "squashing". I was typing on my phone this morning and auto-correct had a sense of humour ;)

This guide is pretty good:
http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html
But if you use a gui tool, doing a rebase would depend on what you use.

Basically squashing is just merging similar commits together to make it cleaner for the pull request. I.e. you can merge the first two commits you made together, as they are both adding comments to the code.

Note that you'll need to do a forced push when you push back to GitHub, as it will need to remove some commits.

kwhitefoot commented:
@Mystro256 I don't think I have time to redo the commits that are already there but I'll certainly bear in mind the idea for the future.

kwhitefoot committed:
Don't store application context in a static field.  Such storage causes memory leaks.

kwhitefoot committed:
Fixed a lot of easy warnings, suppressed warnings that seem unnecessary.

Mystro256 commented:
@kwhitefoot Well you don't have to redo them per-say, squashing just merges commits together, so rather than lots of little commits, you can merge the ones that are related together.

kwhitefoot committed:
Fixed possible nulls by making status local.

kwhitefoot committed:
Make sSyncAdapter an instance field instead of static to remove a potential memory leak.

kwhitefoot committed:
Removed unnecessary if statement.

kwhitefoot committed:
Made some fields final.

kwhitefoot committed:
Acted on some declaration redundancy inspection results.

kwhitefoot committed:
Acted on some declaration redundancy inspection results.

kwhitefoot committed:
Fixed spelling: accountName - accountName, etc.  Reduced visibility of various methods.  Deleted empty onClick method.

kwhitefoot committed:
Made account non-static in SyncAdapter.

kwhitefoot committed:
Begin encapsulating directory names in ImapNotes2Account to reduce repetition.

kwhitefoot committed:
Fixed case of search label -> Search as in Chris Graham's ocControl fork.

kwhitefoot committed:
Added notes about vector clocks and merging.

kwhitefoot committed:
Added SetAccount overloads to hide more of the Data implementation.  Added CreateLocalDirectories and ClearHomeDir to Account class to reduce repetition and put them closer to the data that is used by them.  Moved various declarations nearer to first use.  Upgraded buildtoos to 23.0.3.

kwhitefoot committed:
Implemented Chris Graham's heading change for edit account but added string resource.

kwhitefoot committed:
Implemented Chris Graham's confirmation dialog for deletion.  Once we have the vector clocks and history we can remove this an allow the user to retrieve old versions instead.

kwhitefoot committed:
Implemented Chris Graham's clarification "Select account" -> "Select account to view or edit"

kwhitefoot committed:
Implemented Chris Graham's clarification autosync on save.

kwhitefoot committed:
Implemented Chris Graham's corection for height.

kwhitefoot committed:
Use maxLines=1 instead of singleLine.

kwhitefoot committed:
Fixed null reference error caused by over enthusiastic refactoring.

kwhitefoot committed:
Now using wasabeef rich editor in a branch,

kwhitefoot committed:
Now using wasabeef rich editor in master.  Seems to work so far.  Need to get rid of the Sticky note stuff as it no longer makes sense.

kwhitefoot committed:
Now using wasabeef rich editor in master.  Seems to work so far.  Need to get rid of the Sticky note stuff as it no longer makes sense.

Added flag for AutomaticMerge.  This should probably be an enum together with the stickynotes because they are mutually exclusive.

Added Db and associated classes as a container for the various classes representing tables because we now have two more tables to deal with to support the vector clocks.  Moved NotesDbHelper to the new Db class.

Added MakeMessageWithAttachment function to begin storing messages in attachments instead of inline this should simplify the business of handling composite documents.

Added @NonNull to some argument lists.

kwhitefoot committed:
Fix bug that meant that a failed login showed and empty Toast instead of an error message.  Added the Result type to return both the message and the success flag instead of modifying the statusMessage field which allows the message to be final.

kwhitefoot committed:
Moved ImapNotes2Result instance to local created in return statement.  Now all fields can be final.

kwhitefoot committed:
Moved ImapNotes2Result instance to local created in return statement.  Now all fields can be final.

kwhitefoot committed:
Added icons for the wasabeef rich text editor.  Thank you nbenm for point out the problem.

kwhitefoot committed:
Fixed typo in SQL create table that was causing an uncaught exception and a strange error in the logcat,  See http://stackoverflow.com/questions/2850573/activity-has-leaked-window-that-was-originally-added.

kwhitefoot committed:
Added new constructor to ImapNotes2Account to avoid using setters.  Removed imapnotes2Account field from ImapNotes2 class because it was only used in one call so it can be local to the caller.

kwhitefoot committed:
Added note about travis.

kwhitefoot committed:
Moved Result class to its own file because the same pattern was used elsewhere.  Added function to validate synchronization interval.  Moved creation of progress dialog to LogIn thread as it is the only user.

kwhitefoot committed:
Merge remote-tracking branch 'origin/master'

kwhitefoot committed:
Made fields of ImapNotes2Account public and final, removed the function that mutates the state of the object, use constructors instead.  Get some data from ImapNotes2Account instance instead of using AccountManager because ImapNotes2Account instance should already have them.  Use int instead of long for sync interval.  Added ShowToast function that shows a toast for a caller defined number of seconds.

kwhitefoot committed:
Fixed errors due to removing getters.

kwhitefoot committed:
Fixed bug in GetSynchronizationInterval, forgot to set status = true on success.

kwhitefoot committed:
Replace syncinterval textview with a number picker and removed validation because it is now not possible for it to be invalid.

kwhitefoot committed:
Added notes on device id, started implementing device id.  Removed unnecessary getter, use final field instead.

kwhitefoot committed:
Fixed device id generator bug.
